app/src/main/java/org/oppia/app/player/state/itemviewmodel/StateButtonViewModel.kt

```diff
@@ -0,0 +1,117 @@
+
+  var isNextButtonVisible = ObservableField<Boolean>(false)
+  var isPreviousButtonVisible = ObservableField<Boolean>(false)
+
+  var observableInteractionId = ObservableField<String>()
+  var isInteractionButtonActive = ObservableField<Boolean>(false)
+  var isInteractionButtonVisible = ObservableField<Boolean>(false)
+  var drawableResourceValue = ObservableField<Int>(R.drawable.state_button_primary_background)
+
+  var name = ObservableField<String>()
+
+  fun setObservableInteractionId(interactionId: String) {
+    setNextButtonVisible(false)
+    observableInteractionId.set(interactionId)
+    when (interactionId) {
+      CONTINUE -> {
+        isInteractionButtonActive.set(true)
+        isInteractionButtonVisible.set(true)
+        name.set(context.getString(R.string.state_continue_button))
+        drawableResourceValue.set(R.drawable.state_button_primary_background)
+      }
+      END_EXPLORATION -> {
+        isInteractionButtonActive.set(true)
+        isInteractionButtonVisible.set(true)
+        name.set(context.getString(R.string.state_end_exploration_button))
+        drawableResourceValue.set(R.drawable.state_button_primary_background)
+      }
+      LEARN_AGAIN -> {
+        isInteractionButtonActive.set(true)
+        isInteractionButtonVisible.set(true)
+        name.set(context.getString(R.string.state_learn_again_button))
+        drawableResourceValue.set(R.drawable.state_button_blue_background)
+      }
+      ITEM_SELECT_INPUT, MULTIPLE_CHOICE_INPUT -> {
+        isInteractionButtonActive.set(true)
+        isInteractionButtonVisible.set(false)
+        name.set(context.getString(R.string.state_submit_button))
+        drawableResourceValue.set(R.drawable.state_button_primary_background)
+      }
+      FRACTION_INPUT, NUMERIC_INPUT, NUMERIC_WITH_UNITS, TEXT_INPUT -> {
+        // TODO(#163): The value of isInteractionButtonVisible should be false in this case and it should be updated.
+        //  We are keeping this true for now so that the submit button can work even without any interaction.
+        isInteractionButtonActive.set(true)
+        isInteractionButtonVisible.set(true)
+        name.set(context.getString(R.string.state_submit_button))
+        // TODO(#163): The value of drawable should be R.drawable.state_button_transparent_background as per above explanation.
+        drawableResourceValue.set(R.drawable.state_button_primary_background)
+      }
+    }
+  }
+
+  fun clearObservableInteractionId() {
+    observableInteractionId.set("")
+    isInteractionButtonVisible.set(false)
+    isInteractionButtonActive.set(false)
+  }
+
+  fun setAudioFragmentVisible(isVisible: Boolean) {
+    isAudioFragmentVisible.set(isVisible)
+  }
+
+  fun setNextButtonVisible(isVisible: Boolean) {
+    isNextButtonVisible.set(isVisible)
+  }
+
+  fun setPreviousButtonVisible(isVisible: Boolean) {
+    isPreviousButtonVisible.set(isVisible)
+  }
+
+  fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
```

Comment:
Could we instead handle this via data-binding? If not, I think this function is fine but we should try to de-bounce it. Recomputing the bindings for every character that's typed seems specifically inefficient.

Reply:
This comment makes sense but this function is actually not getting used right, so solving this would in here would not be possible.
For now I am removing this from here and once other interactions are ready it will be introduced as per suggestions.